sources/packages/backend/libs/integrations/src/esdc-integration/student-loan-balances/student-loan-balances.processing.service.ts

```diff
@@ -112,6 +122,15 @@ export class StudentLoanBalancesProcessingService {
             creator: auditUser,
           });
         }
+        if (previousBalanceDate) {
```

Comment:
I am not sure if we need this condition as `previousBalanceDate` could not exist only for the first ever import and also could be handled in `insertZeroBalanceRecords` method.

Reply:
Yes, even if it is only one time we should do it right?
The alternative way would be to merge the query into the other one, which be done, it would increase the complexity of of it, but would still work.
@guru-aot @andrepestana-aot @sh16011993 @katrina-aot-git thoughts?